scripts/scheduler_ingest.py

```diff
@@ -0,0 +1,213 @@
+            else:
+                clean_list.append(item)
+        print(f"New length: {len(clean_list)}")
+        print("END Data clean\n")
+        return clean_list
+
+    @staticmethod
+    def _combine_lists(first, second):
+        """ Combines two lists together.
+        :param first: The first list to be combined
+        :param second: The second list to be combined
+        :return: The single combined list """
+        combined = first + second
+        return combined
+
+    def _sort_by_date(self, data):
+        """ Sorts a list by a date field (specified by the _sort_by_field variable)
+        :param data: The list to be sorted
+        :return: The list sorted by the specified field """
+        return sorted(data, key=lambda date: date[self._sort_by_field])
+
+    @staticmethod
+    def _process_from_separate(cycle_data, maintenance_data):
+        """ Converts pre-processed cycle data items and maintenance day data items
+        into a single list of Cycle objects. Each cycle object contains a list
+        of MaintenanceDay objects - all of which fall within the given cycle period.
+        Maintenance Days which fall outside of cycle periods are highlighted and discarded.
+        :param cycle_data: A pre-processed list containing all valid cycle data items.
+        :param maintenance_data: A pre-processed list of all valid maintenance day data items.
+        :return: A list of Cycle objects containing a list of 0 or more MaintenanceDay objects. """
+        cycle_list = []
+        unused_maintenance_data = maintenance_data.copy()
+        for current_cycle_data in cycle_data:
+
+            cycle_obj = Cycle(current_cycle_data['name'],
+                              current_cycle_data['start'],
+                              current_cycle_data['end'])
+            while len(unused_maintenance_data) > 0:
+                m_day = unused_maintenance_data[0]
+                if m_day['start'] < cycle_obj.start:
+                    # if the next m_day is EARLIER than the current cycle START
+                    if len(unused_maintenance_data) > 1:
+                        cycle_after_string = f"start: {unused_maintenance_data[1].start}, " \
+                                             f"end: {unused_maintenance_data[1].end}"
+                    else:
+                        cycle_after_string = f"NONE (No later cycle dates)"
+                    print(f"WARNING - Encountered maintenance day outside of cycle dates "
+                          f"(assuming cycle_data and maintenance_data "
+                          f"were ordered as earliest-to-latest).\n"
+                          f"Maintenance Day= start: {m_day['start']}, end: {m_day['end']}\n"
+                          f"Closest Cycle before= start: {cycle_obj.start}, end: {cycle_obj.end}\n"
+                          f"Closest Cycle after= {cycle_after_string}\n"
+                          f"This Maintenance Day has therefore been discarded.\n")
```

Comment:
This warning is very useful, but makes the code difficult to read as it breaks it up alot if you are trying to follow the logic. As such could you refactor it into a function for readability?